nav2_waypoint_follower/src/waypoint_follower.cpp

```diff
@@ -261,20 +267,22 @@ WaypointFollower::followWaypoints()
       // Update server state
       goal_index++;
       new_goal = true;
-      if (goal_index >= goal->poses.size()) {
+      if (goal_index == goal->poses.size()) {
```

Comment:
>= in case there's some issues where it increments too fast, the stuff involving action servers and async results are stable right now, but doesn't mean there can't be regressions in rclcpp that this could help protect us from